pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,420 @@
+            } else {
+                subscription_info.hide();
+            }
+        }
+
+        /*
+         * Parses lines like:
+         *
+         * id:  text
+         */
+        function parse_single_subscription(text) {
+            var ret = { };
+            $.each(text.split('\n'), function(i, line) {
+                var pos = line.indexOf(':');
+                if (pos !== -1)
+                    ret[line.substring(0, pos)] = line.substring(pos + 1).trim();
+            });
+            return ret;
+        }
+
+        function parse_multiple_subscriptions(text) {
+            var ret = [ ];
+            var parts = text.split('\n\n');
+            for (var i = 0; i < parts.length; ++i) {
+                var segment = parts[i];
+                if (segment.indexOf('Product Name:') === -1)
+                    continue;
+
+                var segment_info = parse_single_subscription(segment);
+
+                var status = segment_info['Status'];
+
+                /* if we have status details, add those to the status */
+                if (segment_info['Status Details'] !== '')
+                    status = status + ' (' + segment_info['Status Details'] + ')';
+
+                /* convert text output to mustache template variables */
+                ret.push({
+                    'product_name': segment_info['Product Name'],
+                    'product_id': segment_info['Product ID'],
+                    'version': segment_info['Version'],
+                    'arch': segment_info['Arch'],
+                    'status': status,
+                    'starts': segment_info['Starts'],
+                    'ends': segment_info['Ends'],
+                });
+            }
+            return ret;
+        }
+
+        /*
+         * Display information about an action in progress
+         * Make sure we only have one subscription-manager instance at a time
+         */
+        var subscription_manager_running = false;
+        function show_progress_message(message) {
+            is_updating.show();
+            updating_message.text(message);
+        }
+
+        function hide_progress_message() {
+            is_updating.hide();
+        }
+
+        /* since we only call subscription_manager, we only need to check the update message visibility */
+        function action_in_progress() {
+            return is_updating.is(':visible');
+        }
+
+        /* get subscription details using 'subscription-manager list' */
+        function get_subscription_details() {
+            if (action_in_progress()) {
+                status_update_requested = true;
+                return;
+            }
+            status_update_requested = false;
+
+            show_progress_message('Retrieving current subscription details.');
+
+            cockpit.spawn(['subscription-manager', 'list'],
+                    { directory: '/', superuser: true, environ: ['LC_ALL=C'] })
```

Comment:
Suggest adding a ```TODO:``` explaining that this will result in non-translated values, and that we'll need to wait until a DBus API in order to rectify this.